src/main/java/org/jboss/tools/intellij/openshift/tree/application/ApplicationsTreeStructure.java

```diff
@@ -221,34 +219,47 @@ private List<BaseNode<?>> load(Callable<List<BaseNode<?>>> callable, NamespaceNo
             return Collections.singletonList(new MessageNode<>(namespace.getRoot(), namespace, errorMessage));
         }
     }
-    private List<URLNode> getURLs(ComponentNode element) {
-        List<URLNode> results = new ArrayList<>();
-        Odo odo = element.getRoot().getOdo().getNow(null);
-        if (odo == null) {
-            return Collections.emptyList();
-        }
+
+    private List<BaseNode<ComponentNode>> getURLs(ComponentNode componentNode, @NotNull Odo odo) {
+        List<BaseNode<ComponentNode>> nodes = new ArrayList<>();
         try {
-            odo.listURLs(element.getParent().getName(),
-                element.getComponent().getPath(), element.getName()).forEach(url ->
-                    results.add(new URLNode(element, url))
-                );
+            List<URLNode> urls = odo.listURLs(
+                componentNode.getParent().getName(),
+                componentNode.getComponent().getPath(),
+                componentNode.getName())
+              .stream()
+              .map(url -> new URLNode(componentNode, url))
+              .collect(Collectors.toList());
+            if (!urls.isEmpty()) {
+                nodes.addAll(urls);
+            } else {
+                nodes.add(new MessageNode<>(root, componentNode, "<No URLS>"));
```

Comment:
@sbouchet: suggested enhancement: add a node "no URLs" if no urls were found for the component